content/ghc-9.12-cabal-3.14/index.md

```diff
@@ -0,0 +1,126 @@
+You can now apply this mechanism to other typeclasses, like:
+
+```haskell
+default IsString (Text, String)
+```
+
+Which means that in a module with [`OverloadedStrings`][OverloadedStrings], string literals `"like this"` will default to [`Text`][Text] instead of a vague polymorphic type.
+
+### Base Library
+
+* Improvements to backtraces and exceptions. Lots of them.
+
+#### ⚠️ Deprecations
+
+* The deprecation cycle of [GHC.Pack][GHC.Pack] has come to an end. Goodbye!
+* [GHC.Desugar][GHC.Desugar] will be removed in GHC 9.14.
+
+### Code Generation
+
+* Experimental support for the [RISC-V Platform](https://gitlab.haskell.org/ghc/ghc/-/issues/16783);
+* SIMD! In the x86: Most floating-point operations and some integer vector operations are supported 128-bit vectors **without LLVM**. [Get in touch](https://matrix.to/#/#ghc:matrix.org) to help with this effort
+* You can try the new experimental [`-fobject-determinism`][-fobject-determinism] flag to enable deterministic object code generation.
+
+#### ⚠️ Deprecations
+
+* Support 32-bit Windows & macOS/iOS has been dropped;
+* As a result, friendship with the `stdcall` calling convention for FFI is ended. Now `ccall` is your best friend.
+
+You can read the full release notes [here](https://downloads.haskell.org/ghc/9.12.1/docs/users_guide/9.12.1-notes.html).
+
+## Cabal 3.14
+
+### New things
+
+* (Spec v3.14 only) New field: `extra-files` allows you to bundle files in the source distribution (*sdist*) of your cabal package. It serves as an alternative when the files you want to bundle do not fit in the existing `data-files` (for runtime data), `extra-source-files` (built by cabal) or `extra-doc-files` (shipped with Haddocks). This field has no inherent meaning, to avoid misuse of the already existing fields. For instance, you can use `extra-files` to ship `stack.yaml` files, which are not used either by the program at run-time, nor by Cabal or Haddock.
+
+* You can now compile projects dynamically with profiling enabled. The following options are now available:
+  * cabal.project: `profiling-shared: <Boolean>`;
+  * cabal file: `ghc-prof-shared-options` for passing options when building in profiling dynamic way;
+  * Command-line arguments: `--enable-profiling --enable-executable-dynamic`.
+
+* [New GHC options and extensions](https://downloads.haskell.org/ghc/9.12.1/docs/users_guide/9.12.1-notes.html) are supported.
+
+* New build type: Hooks. This build type, intended to eventually replace the Custom build type, integrates better with the rest of the ecosystem (cabal-install, Haskell Language Server).
+
+* The experimental `haddock-project` command supports sub-components.
+
+### Changed
+
+* `cabal init` remembers the chosen language within current session (`Haskell2010`, `GHC2021`, etc.).
+
+* `cabal check` will warn about the insecure `git://` protocol in source-repository.
```

Comment:
As [Oleg noted](https://github.com/haskell/cabal/issues/10659), this is not a warning but an error.